### PR TITLE
#157581722 Create request API

### PR DESCRIPTION
**Pull request by augustineezinwa**
#### What does this PR do?
- This PR creates the request resources for the user
- The request resources are requests created by the user to the admin
- This PR makes sure that  every request is validated and more edge cases are covered before the  creation of any request
#### Description of Task to be completed?
- When an API call is made to `users/requests/` with a `POST` verb, a new request is added to the user archive
- The request has to go through a series of middlewares which are made up of validators.
- These validators are written to cover more edge cases and catch any abnormality from the client side
- Test cases were written to make sure this endpoint reacts appropriately to every possible scenario, be it an invalid request field or missing content
#### How should this be manually tested?
- Clone the repository
- run `npm test` on the command line.
- Also, use postman to test this endpoint
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#157581722
#### Screenshots (if appropriate)
image
image
image

#### Questions:
None

### CODE REVIEW

File: server/controllers/RequestController.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+        * @static
+        * @param {object} req - The request payload sent to the router
+        * @param {object} res - The response payload sent back from the controller
+        * @returns {object} - status Message and the particular request by id.
+        * @description This method gets a request by id in maintenance tracker
+        * @memberOf RequestController
+        */
+  static getRequestById(req, res) {
+    const id = req.params.requestId;
```
Comment (nebanat) [inline server/controllers/RequestController.js]: Is there a reason why you are passing the id as param is instead of request body?

File: server/middlewares/RequestValidator.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    * @static
+    * @param {object} req - The request payload sent to the router
+    * @param {object} res - The response payload sent back from the controller
+    * @param {function} next -The call back function that calls the next router
+    * @returns {object} - status Message and the particular request by id.
+    * @description This method gets a request by id in maintenance tracker
+    * @memberOf RequestController
+    */
+  static validateUrl(req, res, next) {
+    if (Number(req.params.requestId) !== +req.params.requestId) {
```
Comment (nebanat) [inline server/middlewares/RequestValidator.js]: Do well to cover edge cases do you know what happens if I do a Number(string)?
